common/djangoapps/django_sudo/decorators.py

```diff
@@ -0,0 +1,56 @@
+"""
+Custom decorator for django-sudo.
+"""
+from functools import wraps
+
+from sudo.settings import RESET_TOKEN
+from sudo.utils import new_sudo_token_on_activity
+from sudo.views import redirect_to_sudo
+
+
+def sudo_required(func_or_region):
+    """
+    Enforces a view to have elevated privileges.
+    Should likely be paired with ``@login_required``.
+
+    >>> @sudo_required
+    >>> def secure_page(request):
+    >>>     ...
+
+    Can also specify a particular sudo region (to only
+    allow access to that region).
+
+    >>> @sudo_required('admin_page')
+    >>> def secure_admin_page(request):
+    >>>     ...
+    """
```

Comment:
This docstring should mention the addition of `course_id`/`course_key_string`/`library_key_string` parsing to set the region.